src/writer.rs

```diff
@@ -341,14 +410,31 @@ impl<'a, W: Write> ElementWriter<'a, W> {
         Ok(self.writer)
     }
 }
+#[cfg(feature = "serialize")]
+struct ToFmtWrite<T>(pub T);
+
+#[cfg(feature = "serialize")]
+impl<T> std::fmt::Write for ToFmtWrite<T>
+where
+    T: std::io::Write,
+{
+    fn write_str(&mut self, s: &str) -> std::fmt::Result {
+        self.0.write_all(s.as_bytes()).map_err(|_| std::fmt::Error)
+    }
```

Comment:
We should concern about encoding here. But because this is not a single place where we should concern, this can be postponed to next PR.

A possible implementation:
- track written events, intercept `Event::Decl` and extract encoding from it;
- by default assume UTF-8;
- allow to set encoding for `Writer` directly, but after that attempt to write `Event::Decl` with different encoding should fail
- keep copy of encoding (this is a reference under the hood) in the `ToFmtWrite` struct and use it here

Reply:
How about we just document that you shouldn't use `write_serializable` with a non-UTF-8 encoding for now.

Reply:
As far as the other stuff goes, I'd rather not add a bunch of temporary complexity knowing that it will have to be ripped out to finish the encoding work (and I do promise I will get around to finishing that, the only big blocker is async which is a major pain)